Source/Core/BoogiePL.atg

```diff
@@ -1362,11 +1361,10 @@ SpecBlock<out Block/*!*/ b>
   .)
   Ident<out x> ":"
   { LabelOrCmd<out c, out label>
-                       (. if (c != null) {
-                            Contract.Assert(label == null);
+                       (. Contract.Assert(c == null || label == null);
+                          if (c != null) {
                             cs.Add(c);
-                          } else {
-                            Contract.Assert(label != null);
+                          } else if (label != null) {
```

Comment:
This test seems redundant because of the assertion above.

Reply:
No. If `c == null` then the assertion does not tell us anything about `label`. It could be that `label == null` in which case we do not want to enter the branch.